GiniVision/Classes/Core/Helpers/Localization/LocalizableStringResource.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  Localization.swift
+//  GiniVision
+//
+//  Created by Gini GmbH on 7/31/18.
+//
+
+import Foundation
+import UIKit
+
+typealias LocalizationEntry = (value: String, description: String)
+
+protocol LocalizableStringResource {
+    var tableName: String { get }
+    var tableEntry: LocalizationEntry { get }
+    var customizable: Bool { get }
```

Comment:
I could argue that this is not type safe. When you add a new entry to the specific enums, you will have to remember to  also set the `customizable` and possibly the `args` fields. If you forget to do so, the default value would be used, which would often be undesirable.

Since adding new strings is expected, I'd consider this an issue. I'd try to have these fields as associated types so every time someone adds a string, the compiler "reminds" them to also provide a value for the `customizable` and `args` field.

Reply:
I agree with you that this could lead to an issue in the future, but I don't get what you mean in the last paragraph. Every `LocalizableStringResource` implementation is a bunch of strings, and I think it is not possible to add an associatetype for each case.

Nonetheless, I think that removing the `default` declaration inside of `customizable` switch would be enough to make it more visible when adding a new string.

Reply:
The case I don't like is the following:
* I want to add another string
* I add it to one of the enums
* I don't pay attention to the `customizable` field so I don't do anything with it

In this case, my new string uses the default value which might be an error that will only show up at runtime.

I would have been nice to have type safety on that and now allow new enum values without explicitly specifying a `customizable` value.

Reply:
But how can you add type safety to an enum case? What I have is to remove the `default` implementation, so every time you add a new string in the resource, you will have to specify not only the tableEntry but also if it is customizable or not, otherwise it won't compile.